unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuilderProjectsPanel/Scripts/SectionController/SectionsController.cs

```diff
@@ -118,13 +121,29 @@ public void Dispose()
         loadedSections.Clear();
     }
 
-    private void OnOpenSectionRequested(SectionId sectionId)
+    private void OnHideContextMenuRequested()
     {
-        OnRequestOpenSection?.Invoke(sectionId);
+        OnRequestContextMenuHide?.Invoke();
     }
 
-    private void OnHideContextMenuRequested()
+    private void OnUpdateSceneDataRequested(string id, SceneDataUpdatePayload payload)
     {
-        OnRequestContextMenuHide?.Invoke();
+        OnRequestUpdateSceneData?.Invoke(id, payload);
+    }
+
+    private void SubscribeEvents(SectionBase sectionBase)
+    {
+        if (sectionBase is ISectionOpenSectionRequester openSectionRequester)
+        {
+            openSectionRequester.OnRequestOpenSection += OpenSection;
+        }
+        if (sectionBase is ISectionHideContextMenuRequester hideContextMenuRequester)
+        {
+            hideContextMenuRequester.OnRequestContextMenuHide += OnHideContextMenuRequested;
+        }
+        if (sectionBase is ISectionUpdateSceneDataRequester updateSceneDataRequester)
+        {
+            updateSceneDataRequester.OnRequestUpdateSceneData += OnUpdateSceneDataRequested;
```

Comment:
Maybe we should unsubscribe preemptively to avoid multiple subscriptions. Also, is it not necessary to unsubscribe when disposing/destroying?

Reply:
Preemptive unsubscription is not necessary because subscription is only done once: when a new instance of a section is created https://github.com/decentraland/explorer/blob/44ad516d6ea352414b83ae9534fb0e900fe9426d/unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuilderProjectsPanel/Scripts/SectionController/SectionsController.cs#L55-L72
Unsubcribing on dispose is also not necessary. `Dispose` will also Dispose all the instances of the sections created by the SectionsController